src/prefect/engine/executors/dask.py

```diff
@@ -182,13 +193,30 @@ def wait(self, futures: Any) -> Any:
             - Any: an iterable of resolved futures with similar shape to the input
         """
         if self.is_started and hasattr(self, "client"):
-            return self.client.gather(futures)
+            results = self.client.gather(futures)
         elif self.is_started:
             with worker_client(separate_thread=True) as client:
-                return client.gather(futures)
+                results = client.gather(futures)
         else:
             raise ValueError("This executor has not been started.")
 
+        for fut in futures:
+            self.futures.remove(fut)
+
+        return results
+
+    def shutdown(self, wait: bool = True) -> List[Any]:
+        """
+        Signal the executor that it should cancel current pending work, prevent future work from 
+        being submitted, and optionally wait for any currently running futures to finish execution.
+        Args:
+            - wait (bool): wait for any pending work to be completed (defaults to True)
+        """
+        super().shutdown(wait=wait)
+        if self.futures and wait:
```

Comment:
While the executor shutdown is waiting for current work to finish will future work still be able to occur?

Reply:
Oh I see that's what the `self.accepting_work` is used for

Reply:
The only odd thing is that the base class doesn't do anything with `wait`, so it reads kinda odd-like here in a child class